store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/utils/ParquetFilterUtils.java

```diff
@@ -76,347 +77,230 @@
-                                }
-                            } else {
-                                nextSeed = true;
-                            }
-                        } else if (min2seed > 0) {
-                            nextSeed = true;
-                        } else {
-                            if (indexIter.hasNext()) {
-                                indexEntry = indexIter.next();
-                            } else {
-                                indexEntry = null;
-                            }
-                        }
-                    }
+                    buildSeedFilterForIndex(sortedSeeds, ParquetStoreConstants.DESTINATION, group, false, seed2Parts);
                 }
             }
         }
-        return newPathToFilter;
     }
 
     /**
-     * Returns a {@link Tuple2} in which the first entry is a sorted {@link List} of the seeds converted to the form in
+     * Returns a {@link Tuple2} in which the first entry is a sorted {@link Set} of the seeds converted to the form in
      * which they appear in the Parquet files, and the second entry is a {@link Map} from the seeds to a {@link Tuple2}
      * which is <code>null</code> if the seed is an {@link EntitySeed} and consists of the destination vertex and
      * directed type if the seed is an {@link EdgeSeed}.
      *
-     * @param seeds the {@link ElementId}s to query for
      * @param identifier the column that the seed relates to
-     * @param schemaUtils a {@link SchemaUtils} used to get a {@link GafferGroupObjectConverter} to convert the seeds to
-     *                    their equivalent Parquet versions
      * @param group the group that is currently being queried
-     * @param comparator the {@link SeedComparator} used to order the seeds
-     * @return a {@link Tuple2} in which the first entry is a sorted {@link List} of the seeds converted to the form in
+     * @return a {@link Tuple2} in which the first entry is a sorted {@link Set} of the seeds converted to the form in
      * which they appear in the Parquet files, and the second entry is a {@link Map} from the seeds to a {@link Tuple2}
      * which is <code>null</code> if the seed is an {@link EntitySeed} and consists of the destination vertex and
      * directed type if the seed is an {@link EdgeSeed}.
      * @throws SerialisationException if the conversion from the seed to corresponding Parquet objects fails
      */
-    private static Tuple2<ArrayList<Object[]>, HashMap<Object[], Tuple2<Object, DirectedType>>> prepSeeds(
-            final Iterable<? extends ElementId> seeds,
-            final String identifier,
-            final SchemaUtils schemaUtils,
-            final String group,
-            final SeedComparator comparator) throws SerialisationException {
-        final HashMap<Object[], Tuple2<Object, DirectedType>> seed2parts = new HashMap<>();
+    private Tuple2<List<Object[]>, Map<Object[], Tuple2<Object[], DirectedType>>> prepSeeds(final String identifier,
```

Comment:
Do you have to use a Tuple2, can you use a Pair instead as it will be less expensive.